manifests-gen/customizations.go

```diff
@@ -326,3 +334,59 @@ func mergeMaps[K comparable, V any](maps ...map[K]V) map[K]V {
 	}
 	return result
 }
+
+// addInfraClusterProtectionPolicy adds a Validating Admission Policy and Binding for protecting
+// InfraClusters created by the cluster-capi-operator from deletion and editing.
+func addInfraClusterProtectionPolicy(objs []unstructured.Unstructured, providerName string) []unstructured.Unstructured {
+	policy := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "admissionregistration.k8s.io/v1beta1",
+			"kind":       "ValidatingAdmissionPolicy",
+			"metadata": map[string]interface{}{
+				"name": "protect-capi-" + providerName + "cluster",
+			},
+			"spec": map[string]interface{}{
+				"failurePolicy": "Fail",
+				"matchConstraints": map[string]interface{}{
+					"resourceRules": []interface{}{
+						map[string]interface{}{
+							"apiGroups":   []interface{}{"infrastructure.cluster.x-k8s.io"},
+							"apiVersions": []interface{}{"*"},
+							"operations":  []interface{}{"UPDATE", "DELETE"},
+							"resources":   []interface{}{providerName + "clusters"},
+						},
+					},
+				},
+				"validations": []interface{}{
+					map[string]interface{}{
+						"expression": "!(oldObject.metadata.annotations['" + clusterv1.ManagedByAnnotation + "'] == '" + managedByAnnotationValueClusterCAPIOperatorInfraClusterController + "')",
```

Comment:
On delete, do we definitely have `oldObject`?

What happens if the resource does not have `annotations` at all? Please make sure to test these policies

Reply:
> On delete, do we definitely have oldObject?

We do always have `oldObject` for DELETE and UPDATE, it is only null for CREATE, which we don't have in this rule, ref: https://github.com/kubernetes/api/blob/2ed5b3feeb965633f8ec6d42a765eb06e940cafd/admissionregistration/v1alpha1/types.go#L254

> What happens if the resource does not have annotations at all? Please make sure to test these policies

Good point, I thought it would ignore it and validate to false, but it looks like it fails instead, which makes more sense.
I tested things out on the [CEL playground here](https://playcel.undistro.io/?content=H4sIAAAAAAAAA5VTTW%2BcMBD9K64P%2BZBqFliWLpZyinLorWqiHBJ6MGYI7sIY2WaVqOp%2F7wDZTQ%2BrVXvD4zdv3nsefnENHZf801Wr%2FFUPQdUqqEgh2qCCseiv2cVFiZe6G30AF72K3dZHxq56heoFalG9XTKD7FTr3Hnq4rnkZ%2FhK%2FoPd3LAjRmg1GGEHcCpYJww2Th2vLAZnuw5cya%2F5Zz5N%2BorDGMjTzmAt2eP90IKD26XhQ48skTHtYFb0YHrwQfWDpLFpnGYiLkS8fkhimaVykz%2BVfIL%2FZWFup4pk1fJ1xpBk%2F%2Bdk5lutzjKqCfUCOFGRHsmSqYCqB5pmRIfi9WeTuHYndJKnX0SLoUgPED8oTTgSgr41TRAHESRvwjjwdnQaHsH5mbzkSbxOk3yJYTSUq0632yqOM5Er3Yhss9GiKig10HUOcbGNs6Qo0Q%2Bgl6QXe986hXCH9WANhvcMW%2BsDGaJkqBadEB%2Ftey20iWrYHzI5So%2B07ReawTqiybNsPZ1NDRhMePsOzfuYZR3ugd48LJV%2FSMuD24OTbK%2BJjuaSClP1%2BowUWqMwLutBy1XTUwU3Aq1mb2ugrZz%2Bt99%2FAO7T3jB3AwAA) and I came up with a better expression rule.